app/src/hooks/useUnsavedChanges.tsx

```diff
@@ -0,0 +1,83 @@
+import { IYesNoDialogProps } from 'components/dialog/YesNoDialog';
+import { EditSurveyI18N } from 'constants/i18n';
+import { DialogContext } from 'contexts/dialogContext';
+import * as History from 'history';
+import { useContext } from 'react';
+import { useHistory } from 'react-router';
+
+/**
+ * Hook to handle pages that need confirmation before leaving with unsaved changes.
+ * Will render a confirmation dialog when attempting to navigate to a different page.
+ * In most cases this hook will be used in conjunction with the Prompt (react-router-dom) component.
+ *
+ * @param {boolean} [skipConfirmation] - Boolean indicator to skip the dialog confirmation ie: isSaving / isSubmitting
+ * @returns {*} {
+ *  renderUnsavedChangesDialog: (pathname: string) => void - manually trigger the confirmation dialog - usually used with `cancel` callbacks
+ *  changeLocation: (location: History.Location) => boolean - location change interceptor - passed to prompt `message` prop
+ * }
+ */
+export const useUnsavedChangesDialog = (skipConfirmation?: boolean) => {
+  const history = useHistory();
+  const dialogContext = useContext(DialogContext);
+
+  /**
+   * Generates the cancel dialog props
+   *
+   * @param {string} pathname - Path to redirect to on confirmation
+   * @returns {IYesNoDialogProps} Cancel dialog props
+   */
+  const getCancelDialogProps = (pathname: string): IYesNoDialogProps => {
+    return {
+      dialogTitle: EditSurveyI18N.cancelTitle,
+      dialogText: EditSurveyI18N.cancelText,
```

Comment:
We'll need to make this text generic or configurable

Reply:
🌵